### PR TITLE
Add integration tests for F1 help

### PR DESCRIPTION
**Pull request by rchande**
Tag @dotnet/roslyn-ide for review
Will squash when merging.

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 1]
```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
         {
             return view.GetBufferContainingCaret();
         }
+
+        public List<string> GetF1Keyword()
+        {
+            return InvokeOnUIThread(() =>
+            {
+                List<string> results = new List<string>();
+                IVsTextLines textLines;
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: `out var` all the things! (seriously, there's a bunch here)

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 2]
```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
...
+                monitorUserContext.CreateEmptyContext(out emptyUserContext);
+                int line;
+                int column;
+                GetActiveVsTextView().GetCaretPos(out line, out column);
+                var span = new TextSpan();
+                span.iStartLine = line;
+                span.iStartIndex = column;
+                span.iEndLine = line;
+                span.iEndIndex = column;
+                languageContextProvider.UpdateLanguageContext(0, textLines, new[] { span }, emptyUserContext);
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: What is the 0 for?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 3]
```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
...
+                int count;
+                emptyUserContext.CountAttributes("keyword", VSConstants.S_FALSE, out count);
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: What does the `S_FALSE` do?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 4]
```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
         {
             return view.GetBufferContainingCaret();
         }
+
+        public List<string> GetF1Keyword()
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: `ImmutableArray<string>`?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 5]
```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
...
+                Microsoft.VisualStudio.Shell.ServiceProvider.GlobalProvider.QueryService(languageServiceGuid, out languageService);
+                var languageContextProvider = languageService as IVsLanguageContextProvider;
+
+                IVsMonitorUserContext monitorUserContext = GetGlobalService<SVsMonitorUserContext, IVsMonitorUserContext>();
+                IVsUserContext emptyUserContext;
+                monitorUserContext.CreateEmptyContext(out emptyUserContext);
+                int line;
+                int column;
+                GetActiveVsTextView().GetCaretPos(out line, out column);
+                var span = new TextSpan();
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: object initializer?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs [hunk 6]
```diff
@@ -501,5 +502,43 @@ protected override ITextBuffer GetBufferContainingCaret(IWpfTextView view)
...
+                GetActiveVsTextView().GetBuffer(out textLines);
+                Guid languageServiceGuid;
+                textLines.GetLanguageServiceID(out languageServiceGuid);
+                object languageService;
+                Microsoft.VisualStudio.Shell.ServiceProvider.GlobalProvider.QueryService(languageServiceGuid, out languageService);
+                var languageContextProvider = languageService as IVsLanguageContextProvider;
+
+                IVsMonitorUserContext monitorUserContext = GetGlobalService<SVsMonitorUserContext, IVsMonitorUserContext>();
+                IVsUserContext emptyUserContext;
+                monitorUserContext.CreateEmptyContext(out emptyUserContext);
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: Can you observe the results of these calls and give nice failure exceptions if things aren't `S_OK`?

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpF1Help.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    #endregion TaoRegion
+}";
+
+            SetUpEditor(text);
+            Verify("abstract", "abstract_CSharpKeyword");
+            Verify("ascending", "ascending_CSharpKeyword");
+            Verify("from", "from_CSharpKeyword");
+            Verify("First();", "System.Linq.Enumerable.First``1");
+
```
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpF1Help.cs]: weird blankline

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs (hunk 4 above)
Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: GetF1Keywords (plural)?

Comment (dpoeschl) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: (Cyrus has made me not like returning List<T>)

Comment (rchande) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs]: `ImmutableArray` is not marked `Serializable` and so cannot be returned from the inproc part.

**Pull request merged**